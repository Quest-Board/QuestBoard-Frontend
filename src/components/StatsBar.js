import React, { Component } from "react";
import ProgressBar from 'react-bootstrap/ProgressBar'

export default class StatBar extends Component {

    constructor(props){
        super(props);
        this.state = {rank:"King", xp:851}
    }
    

    async componentDidMount(){
        await this.update();
    }

    async update(){
        //event.preventDefault();
        //TODO
        console.log("in update")
        await fetch('https://coms-319-t15.cs.iastate.edu/api/account/stats', {
            method: 'GET',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            }
          })
          .then(response => {
              if (!response.ok){
                alert("Error getting stats")
              }
              return response.json();
            })
          .then(data => {
                this.setState({xp: data.points % 1000});
                switch(data.rank){
                    case 0:
                        this.setState({rank: "Squire"});
                        break;
                    case 1:
                        this.setState({rank: "Knight"});
                        break;
                    case 2:
                        this.setState({rank: "King"});
                        break;
                    default:
                        this.setState({rank: "Squire"});
                        break;
                }
          })
    }


    render() {
        return (
            <div className="stats-bar-wrapper">
                <div id="statsBar" className="stats-bar">
                    <div style={{display: "flex"}}>
                        <img src={require("../images/" + this.state.rank + ".png")} alt={this.state.rank} style={{width:40, alignSelf:"center"}}></img>
                        <h3 style={{alignSelf:"center"}}>{this.state.rank}</h3>
                    </div>
                    <div style={{paddingTop:5}}>
                        <ProgressBar label={`${this.state.xp / 10}%`} now={this.state.xp / 10} />
                    </div>
                </div>
            </div>
            
        );
    }
}
