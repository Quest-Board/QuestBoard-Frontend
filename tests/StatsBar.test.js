import 'regenerator-runtime/runtime';
import React from 'react';
import renderer from 'react-test-renderer';
import StatsBar from '../src/components/StatsBar';


test('test2', () => {
    const component = renderer.create(
        <StatsBar></StatsBar>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});