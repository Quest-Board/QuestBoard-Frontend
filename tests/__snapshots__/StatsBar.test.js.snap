// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test2 1`] = `
<div
  className="stats-bar-wrapper"
>
  <div
    className="stats-bar"
    id="statsBar"
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <img
        alt="Squire"
        src={Object {}}
        style={
          Object {
            "alignSelf": "center",
            "width": 40,
          }
        }
      />
      <h3
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        Squire
      </h3>
    </div>
    <div
      style={
        Object {
          "paddingTop": 5,
        }
      }
    >
      <div
        className="progress"
      >
        <div
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="progress-bar"
          role="progressbar"
          style={
            Object {
              "width": "0%",
            }
          }
        >
          0%
        </div>
      </div>
    </div>
  </div>
</div>
`;
